pkg/scheduler/core/generic_scheduler.go

```diff
@@ -445,9 +445,9 @@ func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) (numNodes i
 	adaptivePercentage := g.percentageOfNodesToScore
 	if adaptivePercentage <= 0 {
 		adaptivePercentage = schedulerapi.DefaultPercentageOfNodesToScore - numAllNodes/125
-		if adaptivePercentage < minFeasibleNodesPercentageToFind {
-			adaptivePercentage = minFeasibleNodesPercentageToFind
-		}
+	}	
+	if adaptivePercentage < minFeasibleNodesPercentageToFind {
+		adaptivePercentage = minFeasibleNodesPercentageToFind
```

Comment:
thinking out loud here...
the current behaviour is this:
* adaptive percentage is set to the `percentageOfNodesToScore`
* If adaptive percentage is less than zero then it calculates a value.
  * If calculated adaptive percentage is less than the minimum feasible number then the adaptive percentage is set to the minimum feasible number.

This patch would do the following:
* adaptive percentage is set to the `percentageOfNodesToScore`
* If adaptive percentage is less than zero then it calculates a value.
* If adaptive percentage is less than the minimum feasible number then the adaptive percentage is set to the minimum feasible number.
 
the minimum feasible is set to 5, which agrees with the docs
> The lower bound for the automatic value is 5%. In other words, the scheduler always scores at least 5% of the cluster no matter how large the cluster is, unless the user provides the config option with a value smaller than 5.

In this case, users could not specify a value lower than 5, which is something that they can do now. This could affect (?) very large clusters (>5k nodes). 
Given that this would change user facing behavior I think it merits a release note.

Reply:
As @alejandrox1 said, we allow users set this value to be lower than 5, it is also mentioned in the [doc](https://kubernetes.io/docs/concepts/scheduling/scheduler-perf-tuning/#percentage-of-nodes-to-score):

> the scheduler always scores at least 5% of the cluster no matter how large the cluster is, unless the user provides the config option with a value smaller than 5

And L448-450 guarantees that `numNodes >= minFeasibleNodesToFind`.

Reply:
I agree. If this is the case, the minimum percentage is only applied to the adaptive percentage calculated from the formula. We then don't need to make a change.  @hex108  @alejandrox1